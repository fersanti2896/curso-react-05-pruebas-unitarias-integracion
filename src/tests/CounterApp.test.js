import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from 'enzyme'
import CounterApp from '../CounterApp';

describe('Pruebas al componente CounterApp', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    })

    test('Debe mostrar a <CounterApp /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe mostrar 100 como valor por defecto en <CounterApp />', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value = {value}/>);
        const valorDef = wrapper.find('h2').text().trim();

        expect(valorDef).toBe('100');
    });

    test('Debe incrementar el contador del botón de +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const valorDef = wrapper.find('h2').text().trim();

        expect(valorDef).toBe('17');
    });

    test('Debe decrementar el contador del botón de -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const valorDef = wrapper.find('h2').text().trim();

        expect(valorDef).toBe('15');
    });

    test('Debe dar por defecto el valor del contador del botón reset', () => {
        const wrapper = shallow(<CounterApp value = {10}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const valorDef = wrapper.find('h2').text().trim();

        expect(valorDef).toBe('10');
    });
});