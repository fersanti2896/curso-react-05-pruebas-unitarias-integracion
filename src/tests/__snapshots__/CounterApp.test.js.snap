// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente CounterApp Debe mostrar a <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    16
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
